lib/Cleantalk/Common/Enqueue/Enqueue.php

```diff
@@ -0,0 +1,318 @@
+<?php
+
+namespace Cleantalk\Common\Enqueue;
+
+use Cleantalk\Templates\Singleton;
+
+class Enqueue
+{
+    use Singleton;
+
+    /**
+     * @var string
+     */
+    protected $plugin_version = '';
+    /**
+     * @var string
+     */
+    protected $assets_path = '';
+    /**
+     * @var string
+     */
+    protected $plugin_path = '';
+    /**
+     * @var string
+     */
+    protected $css_subdir_path = '/css/';
+    /**
+     * @var string
+     */
+    protected $js_subdir_path = '/js/';
+    /**
+     * @var string
+     */
+    private $type = '';
+    /**
+     * @var array
+     */
+    private $errors = array();
+    /**
+     * @var array
+     */
+    private $handles_to_register = array();
+
+    /**
+     * Initializes the Enqueue class.
+     *
+     * @throws \Exception If the plugin version, assets path, or plugin path is not set.
+     */
+    public function init()
+    {
+        if (empty($this->plugin_version)) {
+            throw new \Exception(__('Plugin version is not set.', 'security-malware-firewall'));
+        }
+        if (empty($this->assets_path)) {
+            throw new \Exception(__('Assets path is not set.', 'security-malware-firewall'));
+        }
+        if (empty($this->plugin_path)) {
+            throw new \Exception(__('Plugin path is not set.', 'security-malware-firewall'));
+        }
+        $this->errors = array();
+        add_action('admin_footer', array($this, 'isAllHandlesQueued'), PHP_INT_MAX);
+    }
+
+    /**
+     * Run WP enqueue style/script in accordance with $this->type.
+     *
+     * @param EnqueueDataDTO|null $data
+     */
+    private function load($data)
+    {
+        // Enqueue script logic
+        try {
+            if (is_null($data)) {
+                throw new \Exception(__('Data DTO is invalid', 'security-malware-firewall'));
+            }
+            $this->handles_to_register = array_merge($this->handles_to_register, array($data->handle));
+            // any must have a type
+            if ($this->type === 'css') {
+                wp_enqueue_style($data->handle, $data->web_path, $data->deps, $data->version, $data->media);
+            } elseif ($this->type === 'js') {
+                wp_enqueue_script($data->handle, $data->web_path, $data->deps, $data->version, $data->args);
+            } else {
+                throw new \Exception(__('Unknown type of asset.', 'security-malware-firewall'));
+            }
+        } catch (\Exception $e) {
+            $this->errorLog($e->getMessage());
+        }
+        // output errors to error_log
+        if ($this->hasErrors()) {
+            error_log(__CLASS__ . __(' errors:', 'security-malware-firewall'));
```

Comment:
invalid text domain, probably we need to move this into protected prop for child class

Reply:
Thanks, fixed https://github.com/CleanTalk/wordpress-antispam/pull/587/commits/c0fb5bd585e2e221d1ac10c141668592fc75c4dd